crates/wysiwyg/src/tests/test_to_plain_text.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{dom::to_plain_text::ToPlainText, ComposerModel};
+use widestring::Utf16String;
+
+#[test]
+fn text() {
+    assert_to_plain("abc", "abc");
+    assert_to_plain("abc def", "abc def");
+    // Internal spaces are preserved.
+    assert_to_plain("abc   def", "abc   def");
+}
+
+#[test]
+fn text_with_linebreaks() {
+    // One new line.
+    assert_to_plain(
+        "abc<br />def",
+        r#"abc
+def"#,
+    );
+
+    // Two new lines (isn't transformed into a new block).
+    assert_to_plain(
+        "abc<br /><br />def",
+        r#"abc
+
+def"#,
+    );
+}
+
+#[test]
+fn text_with_italic() {
+    assert_to_plain("<em>abc</em>", "abc");
+    assert_to_plain("abc <em>def</em> ghi", "abc def ghi");
+    assert_to_plain(
+        "abc <em>line1<br />line2<br /><br />line3</em> def",
+        r#"abc line1
+line2
+
+line3 def"#,
```

Comment:
We can use indoc macro to make these nicer to read

e.g.

```rust
        indoc! {
            r#"
            abc line1
            line2
    
            line3 def"#
        },
```

Reply:
I agree, that way is much more readable! [`d62d26e`](https://github.com/matrix-org/matrix-rich-text-editor/pull/573/commits/d62d26e2be63692fe432da1da517b58cad8f7e9c)